conda/gateways/repodata/__init__.py

```diff
@@ -289,3 +290,96 @@ def conda_http_errors(url, repodata_fn):
             e.response,
             caused_by=e,
         )
+
+
+class CacheJsonState(dict):
```

Comment:
I feel like subclassing `dict` and not implementing the parent `__init__` signature seems wrong, could we instead use `collections.UserDict` for this to retain the ability to access a regular dict?

Reply:
```suggestion
class RepodataState(dict):
```
Assuming we use this in the future, it might make sense to forgo the "cache" name and just call it what it is, the representation of the repodata state?